### PR TITLE
JIT: Cosmetic change of `Dim3` class

### PR DESCRIPTION
**Pull request by asi1024**
In the process of cooperative group support (#6620), `Dim3` class was fixed to have two features: as a user interface and as a CUDA type. This PR removes the feature as a user interface from `Dim3` class, and modifies threadIdx and blockDims to be defined as `Data` objects.

cc: @leofang

### CODE REVIEW

**Review by leofang — DISMISSED** [review]

File: cupyx/jit/_internal_types.py [hunk 1]
```diff
@@ -9,11 +9,15 @@ class Expr:
 
 
 class Data(Expr):
-    def __init__(self, code: str, ctype: _cuda_types.TypeBase):
+    def __init__(self, code: str, ctype: '_cuda_types.TypeBase'):
```
[the remark this answers is not in the document]
Comment (asi1024) [reply cupyx/jit/_internal_types.py:12]: This fix was intended to avoid circular imports, but it is not necessary now.

**Review by leofang — APPROVED** [review]

**Comment by kmaehashi** [comment]
I think it's better to keep `x, y, z` attribute descriptions in docs so that users can quickly understand what `threadIdx` is for. LGTM otherwise :)

https://docs.cupy.dev/en/latest/reference/generated/cupyx.jit.threadIdx.html

**Comment by asi1024** [comment]
@kmaehashi Fixed in [952c2a7](https://github.com/cupy/cupy/pull/6644/commits/952c2a73094f7b7d05116d9efe288040c957e9ca)!

**Comment by kmaehashi** [comment]
Looks fine 😄 
https://cupy--6644.org.readthedocs.build/en/6644/reference/generated/cupyx.jit.threadIdx.html

/test mini

**Review by kmaehashi — APPROVED** [review]

**Comment by leofang** [comment]
Thanks, @asi1024!

**Pull request merged**